### DISCUSSION TITLE
vipsheader: reading Orientation from JXL file

### DISCUSSION THREAD

**Discussion by jhdrn** [libvips/libvips#3983] [Help wanted]
Not sure if this is a bug or if I'm doing something wrong, but when running `vipsheader -a my-file.jxl` it always reports an `Orientation` value of `1`, but I expect it to be `8`. I've used Exiftool to read and rewrite the Orientation metadata to the file (`exiftool -Orientation#=8 my-file.jxl`) and vipsheader still does not return the expected Orientation value. 

Does anyone else experience the same issue or am I missing something here?

**Comment by jcupitt** [comment]
Hello @jhdrn,

As far as I know this should work. Do you have a sample image with a non-default orientation?

I think one of the issues here is that orientation can appear in many places: in the XMP, in the EXIF, in the JXL codestream and probably elsewhere too haha. I believe the JXL standard says that codestream values take precedence, so just setting EXIF probably won't work.

Or that's my understanding! It would be worth asking on the JXL discord for clarification and perhaps some sample images for testing. Maybe `cjxl` can set codestream orientation?

**Reply by jhdrn** [reply]
Hi! 

Thanks for the quick reply! You are correct, I found this: https://github.com/libjxl/libjxl/blob/b7961e1318634a95c7d689ec340dbad8a42d4ed4/lib/include/jxl/decode.h#L1251. 

The images I've tested on did not have a non-default rotation and that's what I try to fix :) We're extracting the JXL files from Lightroom exported DNG files and by some reason the orientation is not set (though available in the metadata of the DNG).

I will have to take another approach on this, thanks for pointing me in the right direction!
